ui/src/application/AddApplicationDialog.tsx

```diff
@@ -84,4 +94,14 @@ export default class AddDialog extends Component<IProps, IState> {
         state[propertyName] = event.target.value;
         this.setState(state);
     }
+
+    private handleChangeNumeric(propertyName: string, event: React.ChangeEvent<HTMLInputElement>) {
+        const state = this.state;
+
+        const regex = /^[0-9\b]+$/;
+        if (event.target.value === "" || regex.test(event.target.value)) {
+            state[propertyName] = event.target.value;
+            this.setState(state);
+        }
```

Comment:
```suggestion
        const state = this.state;
        state[propertyName] = parseInt(event.target.value, 10);
        this.setState(state);
```
use parse int instead.

Reply:
This allowed the input to be empty and felt like a better user experience. When using ParseInt, if a user removes all characters from the input field, then NaN would show and the user would be unable to modify it. I found this worked better and then adding the Number() below ensured the proper request to the backend.

Reply:
Yeah, I have nothing against better ux, but the code does set a string to a number property. This should never be done. I've adjusted the code, to a solution which doesn't do this, and thus doesn't require using Number(defaultPriority). Could you take a look over it?

Reply:
With the updates to the number field this looks great!